internal/datastore/postgres/postgres.go

```diff
@@ -134,7 +155,157 @@ type pgDatastore struct {
 	watchBufferLength         uint16
 	revisionFuzzingTimedelta  time.Duration
 	gcWindowInverted          time.Duration
+	gcInterval                time.Duration
+	gcMaxOperationTime        time.Duration
 	splitAtEstimatedQuerySize units.Base2Bytes
+
+	disposed chan struct{}
+	wg       sync.WaitGroup
+}
+
+func (pgd *pgDatastore) runGarbageCollector() {
+	log.Info().Dur("interval", pgd.gcInterval).Msg("garbage collection worker started for postgres driver")
+
+	defer pgd.wg.Done()
+	for {
+		select {
+		case <-pgd.disposed:
+			log.Info().Msg("shutting down postgres GC")
+			return
+		case <-time.After(pgd.gcInterval):
+			err := pgd.collectGarbage()
+			if err != nil {
+				log.Warn().Err(err).Msg("error when attempting to perform garbage collection")
+			} else {
+				log.Debug().Msg("garbage collection completed for postgres")
+			}
+		}
+	}
+}
+
+func (pgd *pgDatastore) getNow(ctx context.Context) (time.Time, error) {
+	// Retrieve the `now` time from the database.
+	nowSQL, nowArgs, err := getNow.ToSql()
+	if err != nil {
+		return time.Now(), err
+	}
+
+	var now time.Time
+	err = pgd.dbpool.QueryRow(datastore.SeparateContextWithTracing(ctx), nowSQL, nowArgs...).Scan(&now)
+	if err != nil {
+		return time.Now(), err
+	}
+
+	// RelationTupleTransaction is not timezone aware
+	// Explicitly use UTC before using as a query arg
+	now = now.UTC()
+
+	return now, nil
+}
+
+func (pgd *pgDatastore) collectGarbage() error {
+	ctx, cancel := context.WithTimeout(context.Background(), pgd.gcMaxOperationTime)
```

Comment:
it seems important to know if GC is timing out frequently - do we want a prom metric for this?

Reply:
I was going to add metrics as a followup, but I can do so in this PR if you like